src/esp/gfx/PbrDrawable.cpp

```diff
@@ -24,20 +24,21 @@ PbrDrawable::PbrDrawable(scene::SceneNode& node,
       lightSetup_{shaderManager.get<LightSetup>(lightSetupKey)},
       materialData_{
           shaderManager.get<MaterialData, PbrMaterialData>(materialDataKey)} {
-  flags_ = PbrShader::Flag::ObjectId;
+  if (materialData_->metallicTexture && materialData_->roughnessTexture) {
+    CORRADE_ASSERT(
+        materialData_->metallicTexture == materialData_->roughnessTexture,
+        "PbrDrawable::PbrDrawable(): if both the metallic and roughness "
+        "texture exist, they must be packed in the same texture based on glTF "
+        "2.0 Spec.", );
+  }
```

Comment:
This check is not enough, especially in case the original material wouldn't have the maps in the expected channels. If you'd use Magnum's material APIs directly, it would be just

```suggestion
CORRADE_ASSERT(materialData_->hasNoneRoughnessMetallicTexture(), "...", );
```

Alternatively you could put this assert directly into the `ResourceManager`, because there you still have access to the `Trade::PbrMetallicRoughnessMaterialData` instance.

Reply:
This sanity check is to only guarantee that when both metallic and roughness texture are specified, they are pointing to the same image. It has to be as long as it was from the glTF file. That is it.

For even stronger check as you suggested, yes, we may put it in the ResourceManager.

Reply:
DONE. Added to the ResourceManager.